### PR TITLE
Patch/ios phonegap fix

### PR DESCRIPTION
**Pull request by Stanley-Okwii**

### CODE REVIEW

File: src/components/CarouselContainer.ts [hunk 1]
```diff
@@ -120,21 +122,25 @@ export default class CarouselContainer extends Component<CarouselContainerProps,
             });
             this.setState({ images, isLoading: false });
         } else if (this.props.dataSource === "XPath" && this.props.imagesEntity) {
```
Comment (djkwagala) [inline src/components/CarouselContainer.ts]: Re-write as L124 to L127
` else if (this.props.dataSource === "XPath" && this.props.imagesEntity && mxObject) {`
`   this.fetchImagesByXPath(mxObject);`
`}`

File: src/components/CarouselContainer.ts [hunk 2]
```diff
@@ -120,21 +122,25 @@ export default class CarouselContainer extends Component<CarouselContainerProps,
...
+    private fetchImagesByXPath(mxObject: mendix.lib.MxObject) {
         const { entityConstraint } = this.props;
         const requiresContext = entityConstraint && entityConstraint.indexOf("[%CurrentObject%]") > -1;
+        const contextGuid = mxObject.getGuid();
         if (!contextGuid && requiresContext) {
             this.setState({ images: [], isLoading: false });
             return;
         }
 
         const constraint = entityConstraint ? entityConstraint.replace("[%CurrentObject%]", contextGuid) : "";
```
Comment (djkwagala) [inline src/components/CarouselContainer.ts]: do recursive replacement of `[%CurrentObject%]` by using regular expressions.
`enityConstraint.replace(/\[%CurrentObject%\]/g, contextGuid)`

**Review by djkwagala — CHANGES_REQUESTED** [review]

File: src/components/CarouselContainer.ts (hunk 1 above)
Comment (Stanley-Okwii) [reply src/components/CarouselContainer.ts]: Done

File: src/components/CarouselContainer.ts (hunk 2 above)
Comment (Stanley-Okwii) [reply src/components/CarouselContainer.ts]: Done

**Pull request merged**